internal/pkg/override/patch.go

```diff
@@ -0,0 +1,268 @@
+		case yaml.DocumentNode:
+			// make sure we are encoding zero into node
+			p.Value = yaml.Node{}
+			return p.encodeAndStop(node)
+		case yaml.MappingNode:
+			for i := 0; i < len(node.Content); i += 2 {
+				if node.Content[i].Value == pointer[0] {
+					node.Content = append(node.Content[:i], node.Content[i+2:]...)
+					return errStopFollowingPointer
+				}
+			}
+		case yaml.SequenceNode:
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			node.Content = append(node.Content[:idx], node.Content[idx+1:]...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (p *yamlPatch) applyReplace(root *yaml.Node) error {
+	if p.Value.IsZero() {
+		return fmt.Errorf("value required")
+	}
+
+	return followJSONPointer(root, p.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) > 0 {
+			return nil
+		}
+		return p.encodeAndStop(node)
+	})
+}
+
+func (p *yamlPatch) encodeAndStop(node *yaml.Node) error {
+	if err := node.Encode(p.Value); err != nil {
+		return err
+	}
+	return errStopFollowingPointer
+}
+
+var errStopFollowingPointer = errors.New("stop following pointer")
+
+type visitNodeFunc func(node *yaml.Node, remaining []string) error
+
+func followJSONPointer(root *yaml.Node, pointer string, visit visitNodeFunc) error {
+	split := strings.Split(pointer, "/")
+	for i := range split {
+		// apply replacements as described https://www.rfc-editor.org/rfc/rfc6901#section-4
+		split[i] = strings.ReplaceAll(split[i], "~1", "/")
+		split[i] = strings.ReplaceAll(split[i], "~0", "~")
+	}
+
+	return followJSONPointerHelper(root, nil, split, visit)
+}
+
+func followJSONPointerHelper(node *yaml.Node, traversed, remaining []string, visit visitNodeFunc) error {
+	if err := visit(node, remaining); err != nil {
+		if errors.Is(err, errStopFollowingPointer) {
+			return nil
+		}
+		return fmt.Errorf("key %q: %w", strings.Join(traversed, jsonPointerSeparator), err)
+	} else if len(remaining) == 0 {
+		return nil
+	}
+
+	switch node.Kind {
+	case yaml.DocumentNode:
+		if len(node.Content) == 0 {
+			return nil // weird, but ok ¯\_(ツ)_/¯
+		}
+
+		return followJSONPointerHelper(node.Content[0], append(traversed, remaining[0]), remaining[1:], visit)
+	case yaml.MappingNode:
+		for i := 0; i < len(node.Content); i += 2 {
+			if node.Content[i].Value == remaining[0] {
+				return followJSONPointerHelper(node.Content[i+1], append(traversed, remaining[0]), remaining[1:], visit)
+			}
+		}
+
+		return fmt.Errorf("key %q: %q not found in map", strings.Join(traversed, jsonPointerSeparator), remaining[0])
+	case yaml.SequenceNode:
+		if remaining[0] == "-" {
+			return followJSONPointerHelper(node.Content[len(node.Content)-1], append(traversed, remaining[0]), remaining[1:], visit)
+		}
```

Comment:
ooh would you mind adding a comment here saying that removing elements using "-" is not allowed according to the spec:
> Note that the use of the "-" character to index an array will always
   result in such an error condition because by definition it refers to
   a nonexistent array element.  

I initially thought that doing the following was a valid option:
```go
- op: remove
  path: /Resources/TaskDef/List/-
```

Reply:
hopefully the updated logic is sufficient; `-` logic only exists in `add` now